### PR TITLE
nginx: upstep to 1.10.0, add Lua module, footprint optimizations

### PR DESCRIPTION
**Pull request by dirkfeytons**
This PR updates nginx to the recently released 1.10.0 (stable branch). It also adds the Lua module. Additionally some tweaks and patches are done to reduce the binary size.
This obsoletes #2101.

Signed-off-by: Dirk Feytons <dirk.feytons@gmail.com>

### CODE REVIEW

**Comment by heil** [comment]
I dont understand why should it be useful to define this
```
+-#define NGX_MAX_PROCESSES         1024
++#define NGX_MAX_PROCESSES         8
```
Can you please clarify?

**Comment by dirkfeytons** [comment]
This define is used by nginx in `./src/os/unix/ngx_process.c` to allocate a static array of `ngx_process_t` structs which it uses to track its worker processes. A value of 1024 is way more than would be useful on OpenWrt devices (standard advice is to have one worker process per CPU core). A `ngx_process_t` is 32 bytes on 32bit systems so this patch saves about 32K of memory.
Perhaps that's not much but since it's a simple patch in code that rarely changes I thought it's still worthwhile to keep.

File: net/nginx/Makefile [hunk 1]
```diff
@@ -186,8 +193,12 @@ ifneq ($(CONFIG_NGINX_HTTP_UPSTREAM_KEEPALIVE),y)
   ADDITIONAL_MODULES += --without-http_upstream_keepalive_module
 endif
 
+TARGET_CFLAGS += -fvisibility=hidden -ffunction-sections -fdata-sections -DNGX_LUA_NO_BY_LUA_BLOCK
```
Comment (dirkfeytons) [inline net/nginx/Makefile]: For the reason behind the `-fvisibility=hidden` see http://mailman.nginx.org/pipermail/nginx-devel/2016-March/008010.html

**Comment by rediceli** [comment]
Should patch auto/endianness depend on global `CONFIG_BIG_ENDIAN` or we always get **BIG ENDIAN**.

* Makefile: 

```patch
PKG_INSTALL:=1

+ifneq ($(CONFIG_BIG_ENDIAN),)
+PLATFORM_ENDIAN=NGX_HAVE_BIG_ENDIAN
+else
+PLATFORM_ENDIAN=NGX_HAVE_LITTLE_ENDIAN
+endif

PKG_CONFIG_DEPENDS := \
... ...
    ( cd $(PKG_BUILD_DIR) ; \
        $(if $(CONFIG_NGINX_LUA),LUA_INC=$(STAGING_DIR)/usr/include LUA_LIB=$(STAGING_DIR)/usr/lib) \
+        PLATFORM_ENDIAN=$(PLATFORM_ENDIAN) \
        ./configure \
            --crossbuild=Linux::$(ARCH) \
            --prefix=/usr \
```

* auto/endianness:

```patch
checking for system byte ordering

END
+if [ -n "$PLATFORM_ENDIAN" ]; then
+    have=$PLATFORM_ENDIAN . auto/have
+else
...... original check stuff
```

**Pull request merged**